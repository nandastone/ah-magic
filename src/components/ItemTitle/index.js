import React from 'react'
import PropTypes from 'prop-types'

const ItemTitle = ({ item }) => {
  return (
    <React.Fragment>
      {item.name}{' '}
      {
        item.stackable > 1
        ? <small className='text-muted'>x{item.stackable}</small>
        : null
      }
    </React.Fragment>
  )
}

ItemTitle.propTypes = {
  item: PropTypes.object.isRequired
}

export default ItemTitle
