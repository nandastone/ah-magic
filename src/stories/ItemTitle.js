import React from 'react'
import { storiesOf } from '@storybook/react'

// Components

import ItemTitle from '../components/ItemTitle'

storiesOf('ItemTitle', module)
  .add('should display the item title', () => (
    <ItemTitle
      name='My Bazooka'
    />
  ))
  .add('should not display the item count if there\'s only one', () => (
    <ItemTitle
      name='My Bazooka'
      count={1}
    />
  ))
  .add('should display the item count when there\'s more than one', () => (
    <ItemTitle
      name='My Bazooka'
      count={10}
    />
  ))
