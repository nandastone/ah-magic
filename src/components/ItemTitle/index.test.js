import React from 'react'
import { shallow } from 'enzyme'
import renderer from 'react-test-renderer'

// Components

import ItemTitle from './'

describe('ItemTitle component', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(<ItemTitle name='My Bazooka' count={50} />)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('should display the item title', () => {
    const wrapper = shallow(<ItemTitle name='My Bazooka' />)
    expect(wrapper.find('[data-test="name"]').text()).toBe('My Bazooka')
  })

  it('should not display the item count if there\'s only one', () => {
    const wrapper = shallow(<ItemTitle name='My Bazooka' count={1} />)
    expect(wrapper.contains('[data-test="count"]')).toBe(false)
  })

  it('should display the item count when there\'s more than one', () => {
    const wrapper = shallow(<ItemTitle name='My Bazooka' count={10} />)
    expect(wrapper.find('[data-test="count"]').text()).toBe('x10')
  })
})
