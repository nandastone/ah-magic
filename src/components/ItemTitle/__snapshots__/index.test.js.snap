// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ItemTitle component renders correctly 1`] = `
Array [
  <span
    data-test="name"
>
    My Bazooka
</span>,
  " ",
  <small
    className="text-muted"
    data-test="count"
>
    x
    50
</small>,
]
`;
